plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -69,6 +80,18 @@ public void done(IJobChangeEvent event) {
     deploy.schedule();
   }
 
+  private IPath createWorkDirectory() throws IOException {
+    String now = Long.toString(System.currentTimeMillis());
+    IPath workDirectory = getTempDir().append(now);
+    Files.createDirectories(workDirectory.toFile().toPath(), new FileAttribute<?>[0]);
+    return workDirectory;
+  }
+
+  private void ensureLoggedInAndSaveCredential(IPath workDirectory, IShellProvider shellProvider) throws IOException, 
```

Comment:
The reason I wanted to emphasize saving the credentials is that it's saved to the work directory and will be used in the subsequent steps, and this is not part of the standard login process. I'll drop `ensure`.